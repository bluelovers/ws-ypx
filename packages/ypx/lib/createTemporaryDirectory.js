"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.newTemporary = exports.createTemporaryDirectory = exports.getCacheDir = void 0;
/**
 * Created by user on 2020/1/28.
 */
const tmp_1 = require("tmp");
const bluebird_1 = __importDefault(require("bluebird"));
const path_1 = require("path");
const cross_spawn_extra_1 = require("cross-spawn-extra");
const fs_extra_1 = require("fs-extra");
function getCacheDir() {
    try {
        let cp = cross_spawn_extra_1.sync('yarn', [
            'config',
            'current',
            '--json',
        ], {
            stripAnsi: true,
            env: process.env,
        });
        let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);
        if (data.tempFolder) {
            return data.tempFolder;
        }
    }
    catch (e) {
    }
    if (process.env.YARN_CACHE_FOLDER && fs_extra_1.pathExistsSync(process.env.YARN_CACHE_FOLDER)) {
        return path_1.join(process.env.YARN_CACHE_FOLDER, '_ypx');
    }
}
exports.getCacheDir = getCacheDir;
function createTemporaryDirectory() {
    return new bluebird_1.default((resolve, reject) => {
        const tmpdir = getCacheDir();
        tmp_1.dir({
            unsafeCleanup: false,
            prefix: 'ypx_',
            dir: tmpdir,
            // @ts-ignore
            tmpdir,
        }, (error, dirPath) => {
            if (error) {
                reject(error);
            }
            else {
                resolve(dirPath);
            }
        });
    });
}
exports.createTemporaryDirectory = createTemporaryDirectory;
async function newTemporary() {
    let tmpDir = await createTemporaryDirectory();
    return {
        get tmpDir() {
            return tmpDir;
        },
        remove() {
            return fs_extra_1.remove(tmpDir);
        }
    };
}
exports.newTemporary = newTemporary;
exports.default = createTemporaryDirectory;
//# sourceMappingURL=createTemporaryDirectory.js.map