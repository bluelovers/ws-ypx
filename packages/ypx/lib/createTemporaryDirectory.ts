/**
 * Created by user on 2020/1/28.
 */
import { dir as tmpDir } from 'tmp';
import Bluebird from 'bluebird';
import { join } from 'path';
import { sync as crossSpawnExtra } from 'cross-spawn-extra';
import { pathExistsSync, remove } from 'fs-extra';

export function getCacheDir(): string
{
	try
	{
		let cp = crossSpawnExtra('yarn', [
			'config',
			'current',
			'--json',
		], {
			stripAnsi: true,
			env: process.env,
		});

		let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);

		if (data.tempFolder)
		{
			return data.tempFolder
		}
	}
	catch (e)
	{

	}

	if (process.env.YARN_CACHE_FOLDER && pathExistsSync(process.env.YARN_CACHE_FOLDER))
	{
		return join(process.env.YARN_CACHE_FOLDER, '_ypx')
	}
}

export function createTemporaryDirectory()
{
	return new Bluebird<string>((resolve, reject) =>
	{
		const tmpdir = getCacheDir();

		tmpDir({
			unsafeCleanup: false,
			prefix: 'ypx_',
			dir: tmpdir,
			// @ts-ignore
			tmpdir,
		}, (error, dirPath) =>
		{
			if (error)
			{
				reject(error);
			}
			else
			{
				resolve(dirPath);
			}
		});
	})
}

export async function newTemporary()
{
	let tmpDir = await createTemporaryDirectory();

	return {
		get tmpDir()
		{
			return tmpDir;
		},
		remove()
		{
			return remove(tmpDir);
		}
	}
}

export default createTemporaryDirectory
