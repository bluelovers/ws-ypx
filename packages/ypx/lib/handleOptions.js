Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.handleOptions = handleOptions;
exports.default = void 0;

var _lodash = require("lodash");

var _path = require("path");

function handleOptions(argv) {
  let result = (0, _lodash.defaultsDeep)((0, _lodash.cloneDeep)(argv), {
    package: [],
    _: [],
    "--": [],
    quiet: false,
    preferOffline: false,
    ignoreExisting: false,
    noInstall: false,
    cwd: argv.cwd ? argv.cwd : process.cwd()
  });

  if (result.userconfig) {
    result.userconfig = (0, _path.resolve)(result.cwd, result.userconfig);
  }

  return result;
}

var _default = handleOptions;
exports.default = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImhhbmRsZU9wdGlvbnMudHMiXSwibmFtZXMiOlsiaGFuZGxlT3B0aW9ucyIsImFyZ3YiLCJyZXN1bHQiLCJwYWNrYWdlIiwiXyIsInF1aWV0IiwicHJlZmVyT2ZmbGluZSIsImlnbm9yZUV4aXN0aW5nIiwibm9JbnN0YWxsIiwiY3dkIiwicHJvY2VzcyIsInVzZXJjb25maWciXSwibWFwcGluZ3MiOiI7Ozs7OztBQUVBOztBQUNBOztBQUVPLFNBQVNBLGFBQVQsQ0FBdUJDLElBQXZCLEVBQ1A7QUFDQyxNQUFJQyxNQUFNLEdBQUcsMEJBQWEsdUJBQVVELElBQVYsQ0FBYixFQUE4QjtBQUMxQ0UsSUFBQUEsT0FBTyxFQUFFLEVBRGlDO0FBRTFDQyxJQUFBQSxDQUFDLEVBQUUsRUFGdUM7QUFHMUMsVUFBTSxFQUhvQztBQUkxQ0MsSUFBQUEsS0FBSyxFQUFFLEtBSm1DO0FBSzFDQyxJQUFBQSxhQUFhLEVBQUUsS0FMMkI7QUFNMUNDLElBQUFBLGNBQWMsRUFBRSxLQU4wQjtBQU8xQ0MsSUFBQUEsU0FBUyxFQUFFLEtBUCtCO0FBUTFDQyxJQUFBQSxHQUFHLEVBQUVSLElBQUksQ0FBQ1EsR0FBTCxHQUFXUixJQUFJLENBQUNRLEdBQWhCLEdBQXNCQyxPQUFPLENBQUNELEdBQVI7QUFSZSxHQUE5QixDQUFiOztBQVdBLE1BQUlQLE1BQU0sQ0FBQ1MsVUFBWCxFQUNBO0FBQ0NULElBQUFBLE1BQU0sQ0FBQ1MsVUFBUCxHQUFvQixtQkFBUVQsTUFBTSxDQUFDTyxHQUFmLEVBQW9CUCxNQUFNLENBQUNTLFVBQTNCLENBQXBCO0FBQ0E7O0FBRUQsU0FBT1QsTUFBUDtBQUNBOztlQUVjRixhIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgSVlQWEFyZ3VtZW50c0lucHV0LCBJWVBYQXJndW1lbnRzIH0gZnJvbSAnLi90eXBlcyc7XG5pbXBvcnQgeyBJVFNQaWNrRXh0cmEsIElUU1JlcXVpcmVkUGljayB9IGZyb20gJ3RzLXR5cGUnO1xuaW1wb3J0IHsgY2xvbmVEZWVwLCBkZWZhdWx0c0RlZXAgfSBmcm9tICdsb2Rhc2gnO1xuaW1wb3J0IHsgcmVzb2x2ZSB9IGZyb20gJ3BhdGgnO1xuXG5leHBvcnQgZnVuY3Rpb24gaGFuZGxlT3B0aW9ucyhhcmd2OiBJWVBYQXJndW1lbnRzSW5wdXQpOiBSZXF1aXJlZDxJWVBYQXJndW1lbnRzPlxue1xuXHRsZXQgcmVzdWx0ID0gZGVmYXVsdHNEZWVwKGNsb25lRGVlcChhcmd2KSwge1xuXHRcdHBhY2thZ2U6IFtdLFxuXHRcdF86IFtdLFxuXHRcdFwiLS1cIjogW10sXG5cdFx0cXVpZXQ6IGZhbHNlLFxuXHRcdHByZWZlck9mZmxpbmU6IGZhbHNlLFxuXHRcdGlnbm9yZUV4aXN0aW5nOiBmYWxzZSxcblx0XHRub0luc3RhbGw6IGZhbHNlLFxuXHRcdGN3ZDogYXJndi5jd2QgPyBhcmd2LmN3ZCA6IHByb2Nlc3MuY3dkKCksXG5cdH0pIGFzIFJlcXVpcmVkPElZUFhBcmd1bWVudHM+O1xuXG5cdGlmIChyZXN1bHQudXNlcmNvbmZpZylcblx0e1xuXHRcdHJlc3VsdC51c2VyY29uZmlnID0gcmVzb2x2ZShyZXN1bHQuY3dkLCByZXN1bHQudXNlcmNvbmZpZyk7XG5cdH1cblxuXHRyZXR1cm4gcmVzdWx0O1xufVxuXG5leHBvcnQgZGVmYXVsdCBoYW5kbGVPcHRpb25zXG4iXX0=