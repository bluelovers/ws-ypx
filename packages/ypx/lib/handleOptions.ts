import { IYPXArgumentsInput, IYPXArguments } from './types';
import { ITSPickExtra, ITSRequiredPick } from 'ts-type';
import { cloneDeep, defaultsDeep } from 'lodash';

export function handleOptions(argv: IYPXArgumentsInput): Required<IYPXArguments>
{
	return defaultsDeep(cloneDeep(argv), {
		package: [],
		_: [],
		"--": [],
		quiet: false,
		preferOffline: false,
		ignoreExisting: false,
		noInstall: false,
		cwd: argv.cwd ? argv.cwd : process.cwd(),
	})
}

export default handleOptions
