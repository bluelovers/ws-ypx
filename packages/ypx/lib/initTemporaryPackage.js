Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.initTemporaryPackage = initTemporaryPackage;
exports.default = void 0;

var _bluebird = _interopRequireDefault(require("bluebird"));

var _fsExtra = require("fs-extra");

var _path = require("path");

var _initConfig = _interopRequireDefault(require("./initConfig"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function initTemporaryPackage(tmpDir) {
  let data = (0, _initConfig.default)();
  return _bluebird.default.all([(0, _fsExtra.writeFile)((0, _path.join)(tmpDir, '.yarnrc'), data.rc), (0, _fsExtra.writeFile)((0, _path.join)(tmpDir, '.yarnrc.yml'), data.yml), (0, _fsExtra.writeFile)((0, _path.join)(tmpDir, 'yarn.lock'), ``), (0, _fsExtra.writeJSON)((0, _path.join)(tmpDir, 'package.json'), {
    "license": "ISC"
  })]);
}

var _default = initTemporaryPackage;
exports.default = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluaXRUZW1wb3JhcnlQYWNrYWdlLnRzIl0sIm5hbWVzIjpbImluaXRUZW1wb3JhcnlQYWNrYWdlIiwidG1wRGlyIiwiZGF0YSIsIkJsdWViaXJkIiwiYWxsIiwicmMiLCJ5bWwiXSwibWFwcGluZ3MiOiI7Ozs7OztBQUFBOztBQUNBOztBQUNBOztBQUNBOzs7O0FBRU8sU0FBU0Esb0JBQVQsQ0FBOEJDLE1BQTlCLEVBQ1A7QUFDQyxNQUFJQyxJQUFJLEdBQUcsMEJBQVg7QUFFQSxTQUFPQyxrQkFBU0MsR0FBVCxDQUFhLENBQ25CLHdCQUFVLGdCQUFLSCxNQUFMLEVBQWEsU0FBYixDQUFWLEVBQW1DQyxJQUFJLENBQUNHLEVBQXhDLENBRG1CLEVBRW5CLHdCQUFVLGdCQUFLSixNQUFMLEVBQWEsYUFBYixDQUFWLEVBQXVDQyxJQUFJLENBQUNJLEdBQTVDLENBRm1CLEVBR25CLHdCQUFVLGdCQUFLTCxNQUFMLEVBQWEsV0FBYixDQUFWLEVBQXNDLEVBQXRDLENBSG1CLEVBSW5CLHdCQUFVLGdCQUFLQSxNQUFMLEVBQWEsY0FBYixDQUFWLEVBQXdDO0FBQ3ZDLGVBQVc7QUFENEIsR0FBeEMsQ0FKbUIsQ0FBYixDQUFQO0FBUUE7O2VBRWNELG9CIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IEJsdWViaXJkIGZyb20gJ2JsdWViaXJkJztcbmltcG9ydCB7IHdyaXRlRmlsZSwgd3JpdGVKU09OIH0gZnJvbSAnZnMtZXh0cmEnO1xuaW1wb3J0IHsgam9pbiB9IGZyb20gXCJwYXRoXCI7XG5pbXBvcnQgYnVpbGRDb25maWcgZnJvbSAnLi9pbml0Q29uZmlnJztcblxuZXhwb3J0IGZ1bmN0aW9uIGluaXRUZW1wb3JhcnlQYWNrYWdlKHRtcERpcjogc3RyaW5nKVxue1xuXHRsZXQgZGF0YSA9IGJ1aWxkQ29uZmlnKCk7XG5cblx0cmV0dXJuIEJsdWViaXJkLmFsbChbXG5cdFx0d3JpdGVGaWxlKGpvaW4odG1wRGlyLCAnLnlhcm5yYycpLCBkYXRhLnJjKSxcblx0XHR3cml0ZUZpbGUoam9pbih0bXBEaXIsICcueWFybnJjLnltbCcpLCBkYXRhLnltbCksXG5cdFx0d3JpdGVGaWxlKGpvaW4odG1wRGlyLCAneWFybi5sb2NrJyksIGBgKSxcblx0XHR3cml0ZUpTT04oam9pbih0bXBEaXIsICdwYWNrYWdlLmpzb24nKSwge1xuXHRcdFx0XCJsaWNlbnNlXCI6IFwiSVNDXCIsXG5cdFx0fSksXG5cdF0pXG59XG5cbmV4cG9ydCBkZWZhdWx0IGluaXRUZW1wb3JhcnlQYWNrYWdlXG4iXX0=