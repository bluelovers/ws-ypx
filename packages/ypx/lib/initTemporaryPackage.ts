import Bluebird from 'bluebird';
import { writeFile, writeJSON } from 'fs-extra';
import { join } from "path";
import buildConfig from './initConfig';

export function initTemporaryPackage(tmpDir: string)
{
	let data = buildConfig();

	return Bluebird.all([
		writeFile(join(tmpDir, '.yarnrc'), data.rc),
		writeFile(join(tmpDir, '.yarnrc.yml'), data.yml),
		writeFile(join(tmpDir, 'yarn.lock'), ``),
		writeJSON(join(tmpDir, 'package.json'), {
			"license": "ISC",
		}),
	])
}

export default initTemporaryPackage
