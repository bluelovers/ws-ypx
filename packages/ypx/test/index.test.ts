/**
 * Created by user on 2020/1/27.
 */

///<reference types="jest"/>

import handleOptions from '../lib/handleOptions';
import createTemporaryDirectory, { newTemporary } from '../lib/createTemporaryDirectory';
import { pathExistsSync } from 'fs-extra';
import { join } from 'path';
import initTemporaryPackage from '../lib/initTemporaryPackage';

test(`handleOptions`, async (done) =>
{
	let actual = handleOptions({
		cwd: '__dirname',
	} as any);

	expect(actual).toHaveProperty('package');
	expect(actual).toHaveProperty('quiet');
	expect(actual).toHaveProperty('preferOffline');
	expect(actual).toMatchSnapshot()

	done();
});

test(`test temp dir`, async (done) =>
{
	let actual = await newTemporary();

	expect(actual.tmpDir).toContain('ypx_');
	expect(pathExistsSync(actual.tmpDir)).toBeTruthy();
	expect(typeof actual.remove).toStrictEqual('function');

	await initTemporaryPackage(actual.tmpDir);

	expect(pathExistsSync(join(actual.tmpDir, '.yarnrc'))).toBeTruthy();
	expect(pathExistsSync(join(actual.tmpDir, '.yarnrc.yml'))).toBeTruthy();
	expect(pathExistsSync(join(actual.tmpDir, 'yarn.lock'))).toBeTruthy();
	expect(pathExistsSync(join(actual.tmpDir, 'package.json'))).toBeTruthy();

	await expect(actual.remove()).resolves.not.toThrow();

	expect(pathExistsSync(actual.tmpDir)).toBeFalsy();

	done();
});
