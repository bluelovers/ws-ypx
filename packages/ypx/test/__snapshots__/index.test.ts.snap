// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleOptions 1`] = `
{
  "--": [],
  "_": [],
  "cwd": "__dirname",
  "ignoreExisting": false,
  "noInstall": false,
  "package": [],
  "preferOffline": false,
  "quiet": false,
}
`;
